docs/en/sql-reference/functions/arithmetic-functions.md

```diff
@@ -6,7 +6,7 @@ sidebar_label: Arithmetic
 
 # Arithmetic Functions
 
-The result type of all arithmetic functions is the smallest type which can represent all possible results. Size promotion happens for integers up to 32 bit, e.g. `UInt8 + UInt16 = UInt32`. If one of the inters has 64 or more bits, the result is of the same type as the bigger of the input integers, e.g. `UInt16 + UInt128 = UInt128`. While this introduces a risk of overflows around the value range boundary, it ensures that calculations are performed quickly using the maximum native integer width of 64 bit.
+The result type of all arithmetic functions is the smallest type which can represent all possible results. Size promotion happens for integers up to 32 bit, e.g. `UInt8 + UInt16 = UInt32`. If one of the inters has 64 or more bits, the result is of the same type as the bigger of the input integers, e.g. `UInt16 + UInt128 = UInt128`. While this introduces a risk of overflows around the value range boundary, it ensures that calculations are performed quickly using the maximum native integer width of 64 bit. Also, this behavior guarantees compatibility with many other databases which provide 64 bit integers (BIGINT) as the biggest integer type.
 
 The result of addition or multiplication of two integers is unsigned unless one of the integers is signed.
```

Comment:
Maybe clarify #52031 also.
Maybe add a link to C++ promotion rules, as CH does it in a similar way

Reply:
To be honest, I am not sure if we do the reader a favor if we link to the C++ promotion rules (e.g. [here](https://en.cppreference.com/w/c/language/conversion) or [here](https://stackoverflow.com/questions/46073295/implicit-type-promotion-rules)). They are 1. quite complex (due to the many alias types and minimum-width types in C++) and 2. we might know C++ quite well but some business analyst aka. Joe Average might have no clue about C++. I'll rather explain these rules in more detail here.

About #52031: Allow me to check that one separately.

Reply:
IMO C++ rules should be irrelevant here and not mentioned.

It's SQL database after all, and it should behave as close as possible to SQL standard or/and like other SQL databases.